### ISSUE TITLE
[2.1.1] - TypeError: _co.layerLoaded is not a function

### ISSUE THREAD

**Issue by Gmarquis45** [hslayers/hslayers-ng#839]
I have got this error using 2.1.1 version with the full example when I try to add a layer directly in code of the appexternal.js file.
It's maybe because it's a group of layer and not a "final" layer.

For example :
- WORKS : 
```
new ol.layer.Image({
        title: "test",
        source: new ol.source.ImageWMS({
          url: 'http://geoservices.brgm.fr/geologie',
          params: {'LAYERS': 'SCAN_F_GEOL1M', 'TILED': true}
        }),
        crossOrigin: 'anonymous'
      }),
```

- DOES NOT WORK :

```
new ol.layer.Image({
        title: "test",
        source: new ol.source.ImageWMS({
          url: 'http://geoservices.brgm.fr/geologie',
          params: {'LAYERS': 'BSS', 'TILED': true}
        }),
        crossOrigin: 'anonymous'
      }),
```

or via "add layer" :
- http://mapsref.brgm.fr/wxs/refcom-brgm/refign? and choose "FONDS_SCAN" **does not work**.
- http://mapsref.brgm.fr/wxs/refcom-brgm/refign? and choose "MONDE_MOD1_1M_FRONTIERE" **works**.

**Comment by raitisbe** [comment]
I made a quickfix version 2.1.2 which solves this, but meanwhile https://github.com/hslayers/hslayers-ng/tree/issue-839 branch addresses it more thoroughly with regards to sublayer toggling on/off. Should be part of the 2.2.0 release @FilipLeitner ?

**Comment by FilipLeitner** [comment]
> I made a quickfix version 2.1.2 which solves this, but meanwhile https://github.com/hslayers/hslayers-ng/tree/issue-839 branch addresses it more thoroughly with regards to sublayer toggling on/off. Should be part of the 2.2.0 release @FilipLeitner ?

Definitely, I'll add missing fixes into #834
